### PR TITLE
Pagination in specification overview

### PR DESCRIPTION
**Pull request by catenoid**
Demo where the page size was set to 1 for a total of 3 services in the database
paging

---
Update: Load pages on scroll (Again, page size is 1)

paging-scroll
---
In `application.properties`, the user can set the page size with `apicenter.pageSize=N`, where it defaults to 10.

### CODE REVIEW

**Comment by fdw** [comment]
I would say we go with 25 services per page and make it not configurable (at first, until someone complains)

**Comment by fdw** [comment]
What do you think about automatically loading the next page when the user's at the end of the list?

File: backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt [hunk 1]
```diff
@@ -60,9 +63,13 @@ class ServiceController @Autowired constructor(
         return specificationIdFromPath
     }
 
-    @GetMapping
-    fun findAllServices(): List<ServiceDto> =
-        serviceHandler.findAll().map { service -> serviceDtoMapper.fromDomain(service) }
+    @GetMapping(params = ["page"])
+    fun findAllServices(@RequestParam("page") page: String): Page<ServiceDto> =
```
[the remark this answers is not in the document]
Comment (catenoid) [reply backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt]: Sorry but I don't understand. I thought this was where the `Page` object was designed for. Could you please be more specific? Is the idea to create a (say) `ApiCenterPage<>` domain entity, with converters to-and-from the Spring `Page<>`? (however there the `Page` object still appears at the controller...)

**Comment by catenoid** [comment]
Potentially throttle the window scroll events ([underscore](https://underscorejs.org/docs/underscore.html#section-85))

File: backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (fdw) [reply backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt]: My concern is a proper abstraction between domain and adapters. Right now we expose a model that has been defined for the DB adapter through the API, and that is not very abstracted ;)
One solution would be just a different model with the necessary fields and mapping between them. Maybe a ´Set<Service>` could already be enough?

File: backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt [hunk 2]
```diff
@@ -60,9 +62,14 @@ class ServiceController @Autowired constructor(
         return specificationIdFromPath
     }
 
-    @GetMapping
-    fun findAllServices(): List<ServiceDto> =
-        serviceHandler.findAll().map { service -> serviceDtoMapper.fromDomain(service) }
+    @GetMapping(params = ["page"])
+    fun findAllServices(@RequestParam(value = "page") page: String): ResultPageDto<ServiceDto> =
+        try {
+            val resultPage = serviceHandler.findAll(page.toInt(), 10).map { service -> serviceDtoMapper.fromDomain(service) }
```
Comment (fdw) [inline backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt]: Could you extract that magic number to somewhere?

File: backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt [hunk 3]
```diff
@@ -60,9 +62,14 @@ class ServiceController @Autowired constructor(
         return specificationIdFromPath
     }
 
-    @GetMapping
-    fun findAllServices(): List<ServiceDto> =
-        serviceHandler.findAll().map { service -> serviceDtoMapper.fromDomain(service) }
+    @GetMapping(params = ["page"])
+    fun findAllServices(@RequestParam(value = "page") page: String): ResultPageDto<ServiceDto> =
```
Comment (fdw) [inline backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt]: I think you can already get that as an `Int`

**Pull request merged**